addons/hr/models/hr_version.py

```diff
@@ -277,73 +277,47 @@ def _unlink_except_last_version(self):
                 )
 
     def write(self, vals):
-        values = vals
         # Employee Versions Validation
-        # ARPI TODO: what if mass edit ?
-        if 'employee_id' in values:
-            if self.filtered(lambda v: len(v.employee_id.version_ids) == 1 and values['employee_id'] != v.employee_id.id):
+        if 'employee_id' in vals:
+            if self.filtered(lambda v: len(v.employee_id.version_ids) == 1 and vals['employee_id'] != v.employee_id.id):
                 raise ValidationError(self.env._("Cannot unassign the only active record of an employee."))
-        if 'active' in values and not values['active']:
+        if 'active' in vals and not vals['active']:
             if self.filtered(lambda v: len(v.employee_id.version_ids) == 1):
                 raise ValidationError(self.env._("Cannot archive the only active record of an employee."))
 
-        if self.env.context.get('sync_contract_dates'):
-            return super().write(values)
+        if self.env.context.get('sync_contract_dates') or "contract_date_start" not in vals and "contract_date_end" not in vals:
+            return super().write(vals)
 
-        if values.get('contract_date_start', False) and len(self.employee_id.version_ids) == 1:
-            values['date_version'] = values['contract_date_start']
+        contract_date_starts = list(filter(bool, self.grouped("contract_date_start").keys()))
+        if not contract_date_starts:
+            return super().write(vals)
+        if len(contract_date_starts) > 1:
+            raise ValidationError(self.env._("Cannot modify multiple versions contract dates with different contracts at once."))
```

Comment:
```suggestion
            raise ValidationError(self.env._("Cannot modify multiple records contract dates with different contracts at once."))
```
Because users are dumb